src/main/kotlin/no/nav/syfo/melding/kafka/producer/PublishAvvistMeldingStatusService.kt

```diff
@@ -0,0 +1,26 @@
+package no.nav.syfo.melding.kafka.producer
+
+import no.nav.syfo.application.database.DatabaseInterface
+import no.nav.syfo.melding.database.domain.PMelding
+import no.nav.syfo.melding.database.domain.toMeldingTilBehandler
+import no.nav.syfo.melding.database.getMeldingerByIds
+import no.nav.syfo.melding.domain.Melding
+import no.nav.syfo.melding.status.database.getUnpublishedAvvistMeldingStatus
+import no.nav.syfo.melding.status.database.updateAvvistMeldingPublishedAt
+
+class PublishAvvistMeldingStatusService(
+    private val database: DatabaseInterface,
+) {
+
+    fun getUnpublishedAvvisteMeldinger(): List<Pair<PMelding.Id, Melding>> {
+        val unpublishedAvvisteMeldingerIds = database.getUnpublishedAvvistMeldingStatus()
+        return database.getMeldingerByIds(unpublishedAvvisteMeldingerIds)
+            .map { Pair(it.id, it.toMeldingTilBehandler()) }
+    }
```

Comment:
`melding` og `melding_status` tabellene er knyttet sammen via `melding_id` feltet, som er `id`-en i `melding`. Når vi da henter `melding_id`-er og gjør disse om til `melding`-er, så går vi bort fra `P`-domenet og mister infoen om IDene. Derfor må vi sende med den tilhørende IDen i et `Pair` i stedet her.

Sånn sett hadde det kanskje vært lettere om foreign-key var `uuid` i stedet for `id` her 🤔

Reply:
Kanskje, men databaseteknisk er det mye bedre å ha foreign-key til id-kolonnen (både plassbesparende og mer effektivt når databasen skal sjekke constraint)